From 184d5c8c38acb6163022fd6b3b2278723b121a68 Mon Sep 17 00:00:00 2001
From: Stephan Mueller <smueller@chronox.de>
Date: Sun, 15 May 2022 16:39:02 +0200
Subject: [PATCH v51 12/25] scheduler - add entropy sampling hook

The scheduler can be used as a source of entropy. This requires the
presence of a hook that invokes the entropy source implementation.

When the scheduler-based entropy source is not compiled, the hook is
folded into a noop which does not affect the scheduler in any way.

Signed-off-by: Stephan Mueller <smueller@chronox.de>
---
 kernel/sched/core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/kernel/sched/core.c b/kernel/sched/core.c
index c52c2eba7c73..beebd0610d7b 100644
--- a/kernel/sched/core.c
+++ b/kernel/sched/core.c
@@ -6,6 +6,7 @@
  *
  *  Copyright (C) 1991-2002  Linus Torvalds
  */
+#include <linux/lrng.h>
 #include <linux/highmem.h>
 #include <linux/hrtimer_api.h>
 #include <linux/ktime_api.h>
@@ -3709,6 +3710,8 @@ ttwu_stat(struct task_struct *p, int cpu, int wake_flags)
 {
 	struct rq *rq;
 
+	add_sched_randomness(p, cpu);
+
 	if (!schedstat_enabled())
 		return;
 
-- 
2.41.0

